pkg/controller/component/component.go

```diff
@@ -189,6 +193,14 @@ func getClusterCompSpec4Component(clusterDef *appsv1alpha1.ClusterDefinition, cl
 			return &cluster.Spec.ComponentSpecs[i], nil
 		}
 	}
+	for _, shardingSpec := range cluster.Spec.ShardingSpecs {
+		shardCompList := sharding.GenShardingCompSpecList(&shardingSpec)
```

Comment:
using sharding functions within the component can easily result in circular dependencies

Reply:
> using sharding functions within the component can easily result in circular dependencies

remove sharding pkg and put it in controllerutil